src/cli/pm_version_command.zig

```diff
@@ -0,0 +1,647 @@
+            try showHelp(ctx, pm, package_json_dir);
+            return;
+        }
+
+        const version_type, const new_version = parseVersionArgument(positionals[1]);
+
+        try verifyGit(package_json_dir, pm);
+
+        var path_buf: bun.PathBuffer = undefined;
+        const package_json_path = bun.path.joinAbsStringBufZ(package_json_dir, &path_buf, &.{"package.json"}, .auto);
+
+        const package_json_contents = bun.sys.File.readFrom(bun.FD.cwd(), package_json_path, ctx.allocator).unwrap() catch |err| {
+            Output.errGeneric("Failed to read package.json: {s}", .{@errorName(err)});
+            Global.exit(1);
+        };
+        defer ctx.allocator.free(package_json_contents);
+
+        const package_json_source = logger.Source.initPathString(package_json_path, package_json_contents);
+        const json = JSON.parsePackageJSONUTF8(&package_json_source, ctx.log, ctx.allocator) catch |err| {
+            Output.errGeneric("Failed to parse package.json: {s}", .{@errorName(err)});
+            Global.exit(1);
+        };
+
+        const scripts = json.asProperty("scripts");
+        const scripts_obj = if (scripts) |s| if (s.expr.data == .e_object) s.expr else null else null;
+
+        if (pm.options.do.run_scripts) {
+            if (scripts_obj) |s| {
+                if (s.get("preversion")) |script| {
+                    if (script.asString(ctx.allocator)) |script_command| {
+                        try RunCommand.runPackageScriptForeground(
+                            ctx,
+                            ctx.allocator,
+                            script_command,
+                            "preversion",
+                            package_json_dir,
+                            pm.env,
+                            &.{},
+                            pm.options.log_level == .silent,
+                            ctx.debug.use_system_shell,
+                        );
+                    }
+                }
+            }
+        }
+
+        const current_version = brk_version: {
+            if (json.asProperty("version")) |v| {
+                switch (v.expr.data) {
+                    .e_string => |s| {
+                        break :brk_version s.data;
+                    },
+                    else => {},
+                }
+            }
+            Output.errGeneric("No version field found in package.json", .{});
+            Global.exit(1);
+        };
+
+        const new_version_str = try calculateNewVersion(ctx.allocator, current_version, version_type, new_version, pm.options.preid, package_json_dir);
+        defer ctx.allocator.free(new_version_str);
+
+        if (!pm.options.allow_same_version and strings.eql(current_version, new_version_str)) {
+            Output.errGeneric("Version not changed", .{});
+            Global.exit(1);
+        }
+
+        {
+            const updated_contents = try updateVersionString(ctx.allocator, package_json_contents, current_version, new_version_str);
```

Comment:
This is not a blocker because I want to get this in for the release, but in a future pr we should change this to use the JSON printer instead. We have code for detecting the indentation and printing with it. Look for `.guess_indentation` for an example of this